ci/date-check/src/main.rs

```diff
@@ -37,13 +38,11 @@ impl fmt::Display for Date {
 
 fn make_date_regex() -> Regex {
     Regex::new(
-        r"(?x) # insignificant whitespace mode
-        <!--\s*
-        [dD]ate:\s*
-        (?P<y>\d{4}) # year
-        -
-        (?P<m>\d{2}) # month
-        \s*-->",
+        r"(?x)
+        (?:<!--\s+date-check:\s+(\D+)\s+(\d{4})\s+-->)
+        |
+        (?:<!--\s+date-check\s+-->\s+(\D+)\s+(\d{4})\b)
```

Comment:
A couple tweaks:

- Added whitespace and comments to make it easier to read
- Switched back to `?P` (named capture groups, should improve the code further down)
- Switched some instances of `\s+` to `\s*` to match the old behavior

```suggestion
        r"(?x) # insignificant whitespace mode
        (<!--\s*
          date-check:\s*
          (?P<m1>\D+)\s+
          (?P<y1>\d{4})\s*-->
        )
        |
        (<!--\s*
          date-check\s*-->\s+
          (?P<m2>\D+)\s+
          (?P<y2>\d{4})\b
        )
```